### PR TITLE
Remove redundant stages

### PR DESCRIPTION
**Pull request by appleton**
- Factor out stage information to a single file.
- Stop suggesting `qa` and `test` stages.

### CODE REVIEW

File: lib/infer.js [hunk 1]
```diff
@@ -1,32 +1,13 @@
...
-      return [app.replace(val.regex,""), val.stage];
-    }
+  const inferredStage = stages.find(stage => stage.inferRegex.test(app));
+
+  if (inferredStage) {
+    return [ app.replace(inferredStage.inferRegex, ''), inferredStage.name ];
   }
-  return [app,"production"];
+
+  return [ app, 'production' ];
```
Comment (gudmundur) [inline lib/infer.js]: I'm assuming that the first element of the returned array is the stem of the app name, if a stage was found. The second element is the stage.

Would you mind refactoring this to use an object instead?

**Comment by gudmundur** [comment]
Minor comment, feel free to ignore.

:+1:

File: lib/infer.js (hunk 1 above)
Comment (appleton) [inline lib/infer.js]: I'm not actually sure and I'd rather not change the interface without knowing what it's supposed to represent so I'm going to keep this as it is and fix later.

File: lib/stages.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const STAGES = [
+  {
+    name: 'review',
+    inferRegex:  /-pr-(\d+)$/
+  },
+  {
+    name: 'development',
+    inferRegex: /-(dev|development)$/
```
Comment (markpundsack) [inline lib/stages.js]: Might I suggest that you map the testing and QA patterns into here too? That way they don't show up as production apps.

Comment (appleton) [inline lib/stages.js]: Good suggestion @markpundsack PRs always welcome and we might even send you a t-shirt ;)

Comment (gudmundur) [inline lib/stages.js]: Hahaha nice! :D

**Pull request merged**